### PR TITLE
drivers/srf02: some optimizations

### PR DESCRIPTION
**Pull request by haukepetersen**
- switched to use xtimer
- simplified interface slightly
- optimized reading of results
- some style cleanups
- updated the documentation

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@haukepetersen the code looks fine to me. Will test next week. If I forget it, just ping me. One thing that came across my mind was the question if we should start taking the bus initialization out of the sensor drivers when touching the code anyway.

File: drivers/srf02/srf02.c [hunk 1]
```diff
@@ -11,91 +12,93 @@
...
-    if (status < 0) {
-        DEBUG("Read the low echo byte from the i2c-interface is failed");
-        distance = UINT16_MAX;
-        return distance;
-    }
-
-    distance = (range_high_byte << 8) | (range_low_byte);
-
-    return distance;
+    /* compile result - TODO: fix for different host byte order other than LE */
```
Comment (PeterKietzmann) [inline drivers/srf02/srf02.c]: Is that really the plan? If so we need to update several drivers I guess...

**Comment by PeterKietzmann** [comment]
Tested and works. ACK in general. Needs rebase

File: drivers/srf02/srf02.c (hunk 1 above)
Comment (haukepetersen) [inline drivers/srf02/srf02.c]: Yes! Most of our drivers can only handle little endian at the moment...

**Comment by haukepetersen** [comment]
Nice, waiting for Travis...

Regarding the bus initialization: I don't want to take it out of the driver for now, as I feel like the discussion on a global concept to this is not quite finished. I'd rather wait for a final solution here and then touch the driver again...

**Comment by PeterKietzmann** [comment]
I agree

**Comment by haukepetersen** [comment]
cool. Now go Travis, go :-)

**Comment by haukepetersen** [comment]
Travis is happy -> and go

**Comment by OlegHahm** [comment]
Feature freeze?

**Comment by PeterKietzmann** [comment]
AFAIK we need that for SAFEST though I gave it a "high" priority

**Comment by kaspar030** [comment]
@haukepetersen Next time please check if switch_to_xtimer already contains xtimer updates.

**Pull request merged**